### ISSUE TITLE
TextBundle Support

### ISSUE THREAD

**Issue by JackalZhao** [gsantner/markor#947]
#### Description

For a long time, inserting images to markdown document is always a problem: 

- Using image bed is not stable. 
- Using local file path is messy for managing files, and also not stable. 

That's what really bothers me when using markdown to write. 

However, `Marked` developer and `Ulysses` team developed a excellent solution: the  **TextBundle**, which is based on **file system**, not *private package format*. May I explain it shortly, it is such kind of a structure: 

- A folder named `Document.textbundle` , includes: 
  - A folder named `assets` , which stores all the relevant media files (like images). 
  - A file named  `info.json` , which stores all the relevant metadata (tags, creating date, author, ......) 
  - A file named  `text.md` , which stores the main document, which uses relevant path to use media files in the  `assets` folder. 

![image](https://user-images.githubusercontent.com/19181833/82756035-13f66d80-9e0a-11ea-828d-da456e848615.png)

It can achieve: including both images and md document within one package, no extra permission requires when previewing, and very convenient for editing directly. 

Every md document and it's relevant images are all stored in one TextBundle package, so that it's very convenient when exporting, copying one passage to another platform or device, every old and new Markdown editor can open, edit, and preview the md file,  so it is compatible with all Markdown editors. 

Softwares such as **Ulysses**, **Bear**, **Zettlr**  or even mind map tools like **Mindnode** are now supporting  **TextBundle** . 

If may support **TextBundle** , we can feel more safe and comfortable when writing. Because we can make sure all of our creation can be exported with all its images and other resources, and we can easily edit these resources using other professional tools. 

Here is the TextBundle [official site](http://textbundle.org/) and the [specifications](http://textbundle.org/spec/). 

When opening a `Document.textbundle` folder, just open the text.md inside instead of treating it as an folder would be a good support. ( Of course there are many other ways to support it better, such as automaticlly copying media files into `asstes` folder )

#### Log

**Comment by gsantner** [comment]
See <https://github.com/gsantner/markor/issues/831>.

And no, I want documents to be centric to itself at Markor. And no metadata files magic.

**Comment by JackalZhao** [comment]
> See #831.
> 
> And no, I want documents to be centric to itself at Markor. And no metadata files magic.

That's fine. 

I communicated with [liandi](https://github.com/88250/liandi) developer, and he said it would be supported on next  [liandi](https://github.com/88250/liandi) v1.1.2 (current is v 1.1.1) . 

LianDi is a great desktop markdown editor designed for the future, so when it's next version comes out, may I recommend it to you. Maybe after experiencing a textbundle implemented editor will make you change some thoughts.  

It is totally fine even if don't support TextBundle, it's just a folder after all. All the image file in the `assets` folder are all compatible with Markor. It is down compatible with all MarkDown editors though.

**Comment by JackalZhao** [comment]
So after a long discussion with the developer of [liandi](https://github.com/88250/liandi) yesterday, we got a simplified resolution. We decided to knock off the JSON magic. 

For each document, like a diary, here is it's structure: 

* a parent folder called `2020-05-27.textbundle` 
* a subfolder called `assets` 
* a document called `text.md` 

That's the simplest structure. The algorism is also simple: 

When opening a folder, check it's name, if the folder name adapts `*.textbundle` , then open the `text.md` inside it, else open and list the folder. And, for images, they will always be copied into the `assets` subfolder. 

Then we can get a document nice and teat. every document with its images and attachments are all in one folder, much easier to manage or copy the images in the future.  

How about that solution for Markor? @gsantner
